tokio/src/sync/rwlock.rs

```diff
@@ -0,0 +1,151 @@
+//! An asynchronous `RwLock`-like type.
+//!
+//! This module provides [`RwLock`], a type that acts similarly to an asynchronous `RwLock`.
+//!
+//! This allows you to do something along the lines of:
+//!
+//! ```rust,no_run
```

Comment:
`no_run` doc tests have a habit of rotting.  I suggest writing the test using `block_on` so it can actually be run.